### PR TITLE
2103 show partner htc art info

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fixes more points of #2103

- Display partner HTC number (currently not working because we don't record a number right now, please ignore for now...)
- Display partner ART number
- Display HTC testing result
- Display HTC testing result date

# Testing
**please use the `2103-show-partner-htc-art-info` branch in the schema repo**

- Enrol a patient A into an ART program and assign an enrolment id
- Enrol patient A into testing program, create a testing encounter (encounter must be in the past) and enter HIV test results
- Go to a second patient B and add a contact tracing entry
- Link patient A to the contact trace of patient B
-  Go to the Tracking Outcome section of the contact tracing form. There should be three new controls displaying the previously entered information about the partner

image

### CODE REVIEW

**Comment by mark-prins** [comment]
Gah! I went through all the steps, set up my database, created encounters.. and there's no extra fields.
Then scrolled down just slightly in the instructions.. and..

```
Please use the 2103-show-partner-htc-art-info branch in the schema repo
```

🙄

**Comment by clemens-msupply** [comment]
> Gah! I went through all the steps, set up my database, created encounters.. and there's no extra fields. Then scrolled down just slightly in the instructions.. and..
> 
> ```
> Please use the 2103-show-partner-htc-art-info branch in the schema repo
> ```
> 
> 🙄

Oh no sorry I put in the screenshot as an afterthought and didn't realized it hid this important information :(

**Review by mark-prins — APPROVED** [review]
🎉 It works!

I had to create a new database and reinitialised etc etc.. but it's all good 🙏 

One thing I find annoying, the 'are you sure you want to navigate away' prompt shows up heaps! can we disable that when it isn't required? 🤔

File: client/packages/programs/src/JsonForms/components/ProgramEvent.tsx [hunk 1]
```diff
@@ -114,13 +130,22 @@ const extractAt = (
...
-  data: string | null | undefined,
+const useDisplayValue = (
+  event: ProgramEventFragment | undefined,
   options?: Options
 ) => {
+  const { customDate } = useFormatDateTime();
+  if (!event) return '';
+
+  if (options?.display?.type === 'eventActiveStartDatetime') {
+    const format = options?.display.format ?? 'P';
```
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply client/packages/programs/src/JsonForms/components/ProgramEvent.tsx:142]: Thank I added a comment

**Pull request merged**